internal/metric/smart_metrics.go

```
@@ -0,0 +1,201 @@
+		}
+	}
+
+	return &data, errors
+}
+
+func getMetrics(device string) (*SmartData, []CustomErr) {
+	cmd := exec.Command("smartctl", "-d", "nvme", "--xall", "--nocheck", "standby", device)
+	out, err := cmd.CombinedOutput()
+
+	// If there's an exit error with exit code 4, we ignore the error
+	if exitErr, ok := err.(*exec.ExitError); ok {
```

golangci-lint (1.64.8)

type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors

(errorlint)